### PR TITLE
Update for evernote v2.0.5

### PR DESCRIPTION
**Pull request by fcastilloec**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://dev.evernote.com/doc/reference/ and https://github.com/evernote/evernote-sdk-js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by PranavSenthilnathan** [comment]
@corps Could you take a look since it looks like almost the whole definition is rewritten.

File: types/evernote/index.d.ts [hunk 1]
```diff
@@ -1,6319 +1,1855 @@
-// Type definitions for evernote v 1.25.8
+// Type definitions for evernote v2.0.5
```
Comment (PranavSenthilnathan) [inline types/evernote/index.d.ts]: Could you enable the `dt-header` in the `tslint.json` file? The version should only be major and minor.

Comment (fcastilloec) [reply types/evernote/index.d.ts]: Sure, I'll enable it. I just didn't even check `tslint.json`. I just notice than a lot of rules are turned off.
When running `dts-gen`, it creates a `tslint.json` with only:

```json
{ "extends": "dtslint/dt.json" }
```
Should I just leave `tslint.json` as above (deleting all those disabled rules)? And, of course, I'll fix any that can be fixed.

File: types/evernote/index.d.ts [hunk 2]
```diff
@@ -1,6319 +1,1855 @@
...
-         *     <li> "NoteAttributes.applicationData.key" - note not found, by key</li>
-         * </ul>
-         */
-        getNoteApplicationDataEntry(guid: string, key: string, cb: Callback<string>): void;
-
-        /**
-         * Update, or create, an entry in the applicationData map for
-         * the note identified by guid.
-         */
-        setNoteApplicationDataEntry(guid: string, key: string, value: string, c
```
Comment (sheetalkamat) [inline types/evernote/index.d.ts]: You dont need module definition here

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Pull request merged**